primitives/mainchain-follower/src/data_source/federated_authority_observation.rs

```diff
@@ -165,6 +170,31 @@ impl FederatedAuthorityObservationDataSourceImpl {
 			for i in 0..keys.len() {
 				let key = keys.get(i).ok_or("Index {i:?} not found in members_map keys")?;
 
+				// Extract the Cardano public key hash from the map key
+				// The key is CBOR-encoded (32 bytes), we need to ditch the first 4 bytes
+				let key_bytes = match key.as_bytes() {
+					Some(bytes) => bytes,
+					None => {
+						log::warn!("Map key at index {} is not bytes, skipping", i);
+						continue;
+					},
+				};
+
+				// Extract 28 bytes for MainchainMember by skipping first 4 bytes
+				if key_bytes.len() != 32 {
+					log::warn!(
+						"Expected 32 bytes for Cardano public key hash, got {}. Skipping.",
```

Comment:
The behaviour here means that keys in the council will be skipped if they are invalid - if e.g. a council of 5 was accidentally set where 3 of the keys were invalid, it would give network control to just two entities

Does this match the spec? Better behaviour might be to invalidate the whole datum.